feat(common): add ReadonlyMap in place of Map in keyValuePipe

ajitsinghkaler opened the pull request:
ReadonlyMap is a superset of Map, in keyValuePipe we do not change the value of the object so ReadonlyPipe Works right in this case and we can accomodate more types. To accomodate more types added ReadonlyMap in Key Value pipe.

Fixes #37308

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Only map support in keyValue pipe

Issue Number: #37308 


## What is the new behavior?
ReadonlyMap support in keyValue pipe

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

petebacondarwin commented on packages/common/src/pipes/keyvalue_pipe.ts:
```diff
@@ -14,7 +14,7 @@ function makeKeyValuePair<K, V>(key: K, value: V): KeyValue<K, V> {
 /**
  * A key value pair.
- * Usually used to represent the key value pairs from a Map or Object.
+ * Usually used to represent the key value pairs from a ReadonlyMap or Object.
```
I would make it clear here that a writeable `Map` is also allowed. Perhaps not even bother to change this documentation??

ajitsinghkaler replied:
@petebacondarwin made the changes

ajitsinghkaler committed:
feat(common): add ReadonlyMap in place of Map in keyValuePipe

ReadonlyMap is a superset of Map, in keyValuePipe we do not change the value of the object so ReadonlyPipe Works right in this case and we can accomodate more types. To accomodate more types added ReadonlyMap in Key Value pipe.

Fixes #37308

ajitsinghkaler commented:
@alxhub @IgorMinar @pkozlowski-opensource @jelbourn can you please review this

jelbourn reviewed (approved):
LGTM, non-breaking as far as I see

Reviewed-for: public-api

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed-for: public-api
Reviewed-for: fw-core

ajitsinghkaler commented:
@IgorMinar global-approvers review is only left

ajitsinghkaler commented:
@petebacondarwin you have already approved yet pull approve is asking for your approval

petebacondarwin reviewed (approved):
Reviewed-for: public-api

petebacondarwin commented:
I had to approve specifically for public-api

AndrewKushnir commented:
[Presubmit](http://test/OCL:322073994:BASE:322074024:1595217868914:1b132ae) + [Global Presubmit](https://test.corp.google.com/OCL:322073994:BASE:322074081:1595217910356:3ea85bf2)
